### DISCUSSION TITLE
WebView and configuration values

### DISCUSSION THREAD

**Discussion by timheuer** [microsoft/vscode-discussions#667] [Extension Development QnA] [answered]
I need to access configuration values from within a webview -- have others done this in a better way than postMessage back and forth?

**Comment by r3m0t** [comment, marked as answer]
I have a single message with all the configurations and send it to the webview initially + whenever `onDidChangeConfiguration` (using `affectsConfiguration` as well). That should suffice unless you are using resource-scoped settings. It's also easier to work with the configuration being one big object with the correct type definitions instead of using method calls.

**Reply by timheuer** [reply]
So are you basically using `vscode.postMessage` back and forth?

**Reply by r3m0t** [reply]
No just from the extension to the webview. So the webview always has the latest settings in a variable. (React state in my case)
